apps/rush-lib/src/cli/scriptActions/GlobalScriptAction.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as os from 'os';
+
+import { BaseScriptAction, IBaseScriptActionOptions } from './BaseScriptAction';
+
+/**
+ * Constructor parameters for GlobalScriptAction.
+ */
+export interface IGlobalScriptActionOptions extends IBaseScriptActionOptions {
+  scriptPath: string;
+}
+
+/**
+ * This class implements custom commands that are run once globally for the entire repo
+ * (versus bulk commands, which run separately for each project).  The action executes
+ * a user-defined script file.
+ *
+ * @remarks
+ * Bulk commands can be defined via common/config/command-line.json.  Rush's predefined "build"
+ * and "rebuild" commands are also modeled as bulk commands, because they essentially just
+ * invoke scripts from package.json in the same way as a custom command.
+ */
+export class GlobalScriptAction extends BaseScriptAction {
+  private _scriptPath: string;
+
+  constructor(
+    options: IGlobalScriptActionOptions
+  ) {
+    super(options);
+    this._scriptPath = options.scriptPath;
+  }
+
+  public run(): Promise<void> {
+    if (!fsx.existsSync(this.rushConfiguration.rushLinkJsonFilename)) {
+      throw new Error(`File not found: ${this.rushConfiguration.rushLinkJsonFilename}` +
+        `${os.EOL}Did you run "rush link"?`);
+    }
```

Comment:
this should be optional

Reply:
After chatting we decided to remove this check entirely.

---
In reply to: [194200168](https://github.com/Microsoft/web-build-tools/pull/684#discussion_r194200168) [](ancestors = 194200168)